core/state/snapshot/generate.go

```diff
@@ -363,7 +364,7 @@ func (dl *diskLayer) generateRange(ctx *generatorContext, trieId *trie.ID, prefi
 		}
 		root, nodes := snapTrie.Commit(false)
 		if nodes != nil {
-			tdb.Update(trie.NewWithNodeSet(nodes))
+			tdb.Update(root, common.Hash{}, trienode.NewWithNodeSet(nodes))
```

Comment:
Here you use `common.Hash{}` as the empty tree root. In other places (statedb.go/1075) you use `types.EmptyRootHash`. I'm assuming these should not be interchangeable? Should we very explicitly enforce one or the other for empty trees? Seems `types.EmptyRootHash` would be the correct one technically.

Same applies throughout the diff.

Reply:
yep, nice catch.